Organizer waitlist

timy-13 opened the pull request:
Create waitlist view for organizer

timy-13 committed:
Initial event waitlist views

(cherry picked from commit 87e16fd8566b51a46a32a0359f7e8927ce1c12ec)

timy-13 committed:
revert browse_profile.xml change i made by accident

timy-13 committed:
query all lists from list manager

timy-13 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into organizer_waitlist

timy-13 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into organizer_waitlist

timy-13 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into organizer_waitlist

timy-13 committed:
Waitlist views

timy-13 committed:
Waitlist unit tests

timy-13 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into organizer_waitlist

timy-13 committed:
Waitlist view fix

timy-13 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into organizer_waitlist

# Conflicts:
#	app/src/main/java/com/example/nachosbusiness/events/Event.java
#	app/src/main/java/com/example/nachosbusiness/events/EventRegistration.java
#	app/src/main/java/com/example/nachosbusiness/events/ListManagerDBManager.java

timy-13 committed:
Implement waitlist views and lottery sampling

timy-13 committed:
Fix invited/accepted/cancel list queries to retrieve users properly

aabrady6 commented on app/src/main/java/com/example/nachosbusiness/events/Event.java:
```diff
@@ -1,11 +1,12 @@
 package com.example.nachosbusiness.events;
 
-import com.example.nachosbusiness.utils.QRUtil;
```
This needs to import from utils.qrutil

aabrady6 commented on app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java:
```diff
@@ -0,0 +1,232 @@
+
+        ImageButton backButton = view.findViewById(R.id.back);
+        backButton.setOnClickListener(v -> {
+            Intent intent = new Intent(getActivity(), Dashboard.class);
```
App crashes here

aabrady6 commented on app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java:
```diff
@@ -0,0 +1,232 @@
+                    } else if (count < 0) {
+                        Toast.makeText(getContext(), "Sample must be greater than zero!", Toast.LENGTH_SHORT).show();
+                    } else {
+                        sampleCount = count;
```
App crashes if no value is put in. Prob make this a required field.

aabrady6 commented on app/src/main/java/com/example/nachosbusiness/organizer_views/WaitlistFragment.java:
```diff
@@ -0,0 +1,232 @@
+                        return;
+                    }
+
+                    int count = Integer.parseInt(countInput.getText().toString());
```
maybe add some limits on input values here? Can crash app if put in big number like seth did on the demo

timy-13 committed:
Add error checks to sampling

timy-13 committed:
Change sampling to use attendee spots determined at event creation

timy-13 committed:
Fixed list update issue

timy-13 committed:
Fix list update issue

timy-13 committed:
Add profile images to waitlist

timy-13 committed:
add moveAllToCanceledList

timy-13 committed:
add functionality to select which lists to display

timy-13 committed:
cleanup

timy-13 committed:
fix canceled user status

timy-13 committed:
Merge branch 'refs/heads/main' into organizer_waitlist

timy-13 commented:
Recent changes:
* Fixed the weird list db bug that aaron discovered
* Updated UI to show profile pictures and look a bit nicer
* Added radio buttons to choose which lists to display
* Changed the sampling to use the value predetermined at event creation

timy-13 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into organizer_waitlist

# Conflicts:
#	app/src/main/java/com/example/nachosbusiness/events/Event.java
#	app/src/main/java/com/example/nachosbusiness/events/ListManagerDBManager.java

timy-13 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into organizer_waitlist
